sympy/physics/vector/point.py

```diff
@@ -480,22 +480,41 @@ def vel(self, frame):
         frame : ReferenceFrame
             The frame in which the returned velocity vector will be defined in
 
+        otherpoint : Point
+            The point w.r.t which self's position is defined, works as initial
+            point.
+
         Examples
         ========
 
-        >>> from sympy.physics.vector import Point, ReferenceFrame
+        >>> from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
         >>> N = ReferenceFrame('N')
         >>> p1 = Point('p1')
         >>> p1.set_vel(N, 10 * N.x)
         >>> p1.vel(N)
         10*N.x
+        >>> p = Point('p')
+        >>> q = dynamicsymbols('q')
+        >>> p.set_vel(N, 10 * N.x)
+        >>> p2 = Point('p2')
+        >>> p2.set_pos(p, q*N.x)
+        >>> p2.vel(N)
+        (Derivative(q(t), t) + 10)*N.x
```

Comment:
Does this work?

```suggestion
        >>> p = Point('p')
        >>> q = dynamicsymbols('q')
        >>> r = dynamicsymbols('r')
        >>> p.set_vel(N, 10 * N.x)
        >>> p2 = Point('p2')
        >>> p2.set_pos(p, q*N.x)
        >>> p3 = Point('p3')
        >>> p3.set_pos(p2, r*N.x)
        >>> p3.vel(N)
        (Derivative(q(t), t) + Derivative(r(t), t) + 10)*N.x
```

Reply:
Yeah got it

Reply:
perhaps update _vel_dict inside set_pos. -- It can't be done I guess as set_pos has nothing to do with reference frame.

Reply:
You're right; it also can't be done that way, because it wouldn't work for
```
        >>> p2 = Point('p2')
        >>> p2.set_pos(p, q*N.x)
        >>> p3 = Point('p3')
        >>> p3.set_pos(p2, r*N.x)

        >>> p.set_vel(N, 10 * N.x)  # do this last

        >>> p3.vel(N)
        (Derivative(q(t), t) + Derivative(r(t), t) + 10)*N.x
```